### ISSUE TITLE
taskbar doesn't show up when a background window is intersecting with taskbar

### ISSUE THREAD

**Issue by ab-c-d** [ChanpleCai/SmartTaskbar#118]
**Describe the bug**
Auto mode is broken - taskbar doesn't show up when a background window is intersecting with taskbar

**To Reproduce**
Open a maximized file explorer window
Open another file explorer window so that it does not intersect with the taskbar
Focus the second window

**Expected behavior**
Taskbar is visible

**Desktop (please complete the following information):**
 - OS: Windows 11 version 21H2 os build 22000.613
 - Version v1.4.4

**Additional context**
It was working correctly but suddenly started to misbehave recently.

**Comment by ChanpleCai** [comment]
Oh, this logic was just changed a while ago. Because the right-click menu of some applications will also display the taskbar. Some people find this annoying, so I temporarily changed it to this. It may be made optional later, (not sure how many people prefer the original way).

**Comment by ab-c-d** [comment]
The original way is exactly why I am using this program.
>> Because the right-click menu of some applications will also display the taskbar.

I have never seen a program like that.

Which is the latest version that has the original version? Also is there a way to prevent auto-update?

**Comment by ChanpleCai** [comment]
Downloading from GitHub will not automatically update, and the Github version can avoid the taskbar being automatically hidden. 

It should be a day or two. I'll post it after testing for a while. The store version will be a week or two behind.

**Comment by ab-c-d** [comment]
The latest version downloaded from GitHub still has the undesirable behavior.

Could you know which release has the original behavior?

**Comment by ChanpleCai** [comment]
try this: [v1.4.5](https://github.com/ChanpleCai/SmartTaskbar/releases/download/v1.4.5/SmartTaskbar_Setup.exe)

**Comment by ab-c-d** [comment]
Thank you!

Btw, I noticed that sometimes the taskbar become immediately visible after unnecessary auto-hide animation is played, I guess this is what the last updates tried to fix? Is it something that you've described in readme?
> Auto Mode is based on Auto-Hide mode...

Steps to reproduce

 - Open 2 file explorer windows, one intersecting with the taskbar and the other isn't
 - Minimize both
 - Bring the intersecting one to the foreground
   - This makes the taskbar to be hidden
 - Bring the other one to the foreground
   - Then the taskbar auto-hide animation is played, and then it is brought up immediately

**Comment by Oliviaophia** [comment]
Yes, when the two focused windows are switched, especially when you click the taskbar icon with the mouse, Windows will hide the taskbar first when the mouse leaves the taskbar. and I haven't found a way to avoid the automatic hiding.
Previous versions avoided the auto-hide behavior of the taskbar when the start menu was closed, the search box was closed, and the form was closed. However, the hidden behavior of the taskbar caused by clicking the taskbar icon with the mouse is not among them.
This logic may be integrated inside the taskbar, and in that case it may never be avoided.

**Comment by Oliviaophia** [comment]
The current readme is not well organized. I'm just too lazy to do this work.
